honeybee/room.py

```diff
@@ -0,0 +1,605 @@
+# coding: utf-8
+"""Honeybee Room."""
+from .properties import RoomProperties
+from .face import Face
+from .shade import Shade
+from .facetype import get_type_from_normal, Wall, Floor
+from .boundarycondition import get_bc_from_position, Outdoors, Surface
+from .typing import valid_string, float_in_range
+import honeybee.writer as writer
+
+from ladybug_geometry.geometry2d.pointvector import Vector2D
+from ladybug_geometry.geometry3d.pointvector import Vector3D, Point3D
+from ladybug_geometry.geometry3d.plane import Plane
+from ladybug_geometry.geometry3d.polyface import Polyface3D
+
+import math
+
+
+class Room(object):
+    """A volume enclosed by faces, representing a single room or space.
+
+    Properties:
+        name
+        name_original
+        faces
+        indoor_shades
+        outdoor_shades
+        geometry
+        center
+        volume
+        floor_area
+        exposed_area
+        exterior_wall_area
+        exterior_aperture_area
+        average_floor_height
+    """
+    __slots__ = ('_name', '_name_original', '_geometry', '_faces',
+                 '_indoor_shades', '_outdoor_shades', '_properties')
+
+    def __init__(self, name, faces):
+        """A volume enclosed by faces, representing a single room or space.
+
+        Note that using the base initialization method means that checks will not
+        be performed to determine whether the room is a closed volume or to flip
+        faces in the event that they are not facing outward from a room. In cases
+        where these checks are important, the from_faces method should be used.
+
+        Args:
+            name: Room name. Must be < 100 characters.
+            faces: A list of honeybee Face objects that together form the closed
+                volume of a room.
+        """
+        # process the name
+        self._name = valid_string(name, 'honeybee room name')
+        self._name_original = name
+
+        # process the geometry
+        self._faces = []
+        for face in faces:
+            assert isinstance(face, Face), \
+                'Expected honeybee Face. Got {}'.format(type(face))
+            face._parent = self
+            self._faces.append(face)
+        self._geometry = None  # will be calculated from faces or added by classmethods
+        self._indoor_shades = []  # shades can be assigned later
+        self._outdoor_shades = []  # shades can be assigned later
+
+        # initialize properties for extensions
+        self._properties = RoomProperties(self)
+
+    @classmethod
+    def from_faces(cls, name, faces, tolerance, angle_tolerance):
```

Comment:
The main reason for this not a performance issue.  Primarily, it's that, if we force this to be the method that everyone has to use, it won't be be impossible to create a Room object if you don't know the tolerance under which the geometry was originally created.  This makes Rooms the only geometry object that is tied to a tolerance and it means that we loose the opportunity for the honeybee-core library to accept geometry from sources where the original tolerance has been lost.  If we are certain that we only want honeybee-core to accept geometry from certain sources, we can go this route but I have been hoping that honeybee-core is destined for more inter-operability than that.

Reply:
Thinking about it more, I could move the two tolerance arguments into init and just set them equal to None by default, which will effectively do the same thing that the init method does now. And, if someone inputs a numerical tolerance, it will run the solidity checks like the current from_faces method. I had originally thought that having two separate methods to initialize the Room was more explicit but maybe you are right that having users recognize the tolerance as None is better. Let me know your thoughts.